src/qibo/quantum_info/metrics.py

```diff
@@ -590,6 +607,89 @@ def bures_distance(state, target, check_hermitian: bool = False, backend=None):
     return distance
 
 
+def entanglement_fidelity(
+    channel, nqubits: int, state=None, check_hermitian: bool = False, backend=None
+):
+    """Entanglement fidelity of a ``channel`` :math:`\\mathcal{E}` on ``state``
+    :math:`\\rho`, which is given by
+
+    .. math::
+        \\begin{align*}
+          F_{\\mathcal{E}} &= \\text{fidelity}(\\rho_{f}, \\rho) \\nonumber \\\\
+          &= \\tr(\\rho_{f} \\, \\rho)
+
+    where
+
+    .. math::
+        \\rho_{f} = \\mathcal{E}_{A} \\otimes I_{B}(\\rho)
+
+    is the state after the channel :math:`\\mathcal{E}` was applied to
+    partition :math:`A`.
+
+    Args:
+        channel (:class:`qibo.gates.channels.Channel`): quantum channel
+            acting on partition :math:`A`.
+        nqubits (int): total number of qubits in ``state``.
+        state (ndarray, optional): statevector or density matrix to be evolved
+            by ``channel``. If ``None``, defaults to the maximally entangled state
+            :math:`\\frac{1}{2^{n}} \\, \\sum_{k} \\, \\ket{k}\\ket{k}`, where
+            :math:`n` is ``nqubits``. Defaults to ``None``.
+        check_hermitian (bool, optional): if ``True``, checks if the final state
+            :math:`\\rho_{f}` is Hermitian. If ``False``, it assumes it is Hermitian.
+            Defaults to ``False``.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses :class:`qibo.backends.GlobalBackend`.
+            Defaults to ``None``.
+
+    Returns:
+        float: Entanglement fidelity :math:`F_{\\mathcal{E}}`.
+    """
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if isinstance(nqubits, int) is False:
+        raise_error(
+            TypeError, f"nqubits must be type int, but it is type {type(nqubits)}."
+        )
+
+    if nqubits <= 0:
+        raise_error(
+            ValueError, f"nqubits must be a positive integer, but it is {nqubits}."
+        )
+
+    if state is not None and (
+        (len(state.shape) not in [1, 2])
+        or (len(state) == 0)
+        or (len(state.shape) == 2 and state.shape[0] != state.shape[1])
+    ):
+        raise_error(
+            TypeError,
+            f"state must have dims either (k,) or (k,k), but have dims {state.shape}.",
+        )
+
+    if isinstance(check_hermitian, bool) is False:
+        raise_error(
+            TypeError,
+            f"check_hermitian must be type bool, but it is type {type(check_hermitian)}.",
+        )
+
+    if state is None:
+        state = backend.plus_state(nqubits)
```

Comment:
Now that all states should be transformed to density matrices, I think it would be better to generate a density matrix here already.
```suggestion
        state = backend.plus_density_matrix(nqubits)
```